File: contests/february challenge 2019/01. Appy and Contest.py
def gcd(a, b):
	if(b==0):
		return a
	return gcd(b, a%b)

t=int(raw_input())
while(t>0):
	n, a, b, k = map(int, raw_input().split())
	x=n/a
	y=n/b
	z=(n*gcd(a,b))/(a*b)
	# print(x, y, z)
	if(x+y-(2*z))>=k:
		print("Win")
	else:
		print("Lose")
	t-=1

File: contests/february challenge 2019/01. Appy and Contest.cpp
//did not work for larger test cases -> shifted to python.

#include <iostream>
#include <boost/math/common_factor.hpp> 
using namespace std;

int main(int argc, char const *argv[])
{
	int t;
	cin>>t;
	while(t-->0){
		int n, a, b, k;
		cin>>n; 
		cin>>a; 
		cin>>b;
		cin>>k;

		int x=n/a;
		int y=n/b;
		int z=n/boost::math::lcm(a,b);

		if(x+y-(2*z)>=k)
			cout<<"Win\n";
		else
			cout<<"Lose\n";

	}
	return 0;
}